### PR TITLE
Add configuration and training script for transformer model

### PR DESCRIPTION
**Pull request by KKKUBAKKK**
This pull request introduces a new training pipeline for a transformer-based model, including configuration management, model initialization, and training logic. The changes focus on modularizing the configuration setup and implementing a robust training loop with support for checkpointing and TensorBoard logging.

### Configuration Management:

* Added a `get_config` function in `config.py` to centralize model configurations, including hyperparameters (e.g., batch size, learning rate, sequence length) and file paths for weights and tokenizer. (`config.py`, [config.pyR1-R30](diffhunk://#diff-117426151e93a626f8b46bfdb3a95b3f4a62e5f4dd6e65975a7c50759bf04482R1-R30))
* Implemented a `get_weights_file_path` function in `config.py` to dynamically generate file paths for saving and loading model weights. (`config.py`, [config.pyR1-R30](diffhunk://#diff-117426151e93a626f8b46bfdb3a95b3f4a62e5f4dd6e65975a7c50759bf04482R1-R30))

### Training Logic:

* Introduced a `train_model` function in `train.py` to handle the training process, including device setup, data loading, model initialization, and training loop. (`train.py`, [train.pyR1-R90](diffhunk://#diff-ed183d67207df065a11e1289f19d34cc2abbc5448dea952683cfe9728c342b95R1-R90))
* Added support for model checkpointing by saving model state, optimizer state, and epoch information after each epoch. (`train.py`, [train.pyR1-R90](diffhunk://#diff-ed183d67207df065a11e1289f19d34cc2abbc5448dea952683cfe9728c342b95R1-R90))
* Integrated TensorBoard logging to track training loss over time for better monitoring and debugging. (`train.py`, [train.pyR1-R90](diffhunk://#diff-ed183d67207df065a11e1289f19d34cc2abbc5448dea952683cfe9728c342b95R1-R90))

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+def get_config():
+    """
+    Get the configuration for the model
+    """
+    return {
+        "batch_size": 8,
+        "num_epochs": 20,
+        "lr": 1e-4,
+        "seq_len": 350,
```
Comment (KwiatkowskiML) [inline config.py:12]: There is a constant in constants.py representing d_model

File: train.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import warnings
+from pathlib import Path
+
```
Comment (KwiatkowskiML) [inline train.py:3]: missing import torch.nn as nn

File: train.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import torch
+from sympy.physics.quantum import state
+from torch.utils.tensorboard import SummaryWriter
+
+from config import get_weights_file_path
+
+
+def get_model(config, vocab_src_len, vocab_tgt_len):
```
Comment (KwiatkowskiML) [inline train.py:11]: config shadows name from outer space

File: train.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    """
+    Build the model
+    """
+    model  = build_transformer(vocab_src_len, vocab_tgt_len, config['seq_len'], config['seq_len'], config['d_model'])
```
Comment (KwiatkowskiML) [inline train.py:15]: build_transformer is not imported

File: train.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    writer = SummaryWriter(config["experiment_name"])
+
+    optimizer = torch.optim.Adam(model.parameters(), lr=config['lr'], eps=1e-9)
+
+    initial_epoch = 0
+    global_step = 0
+    if config['preload']:
+        model_filename = get_weights_file_path(config, config['preload'])
+        print(f'Preloading model from {model_filename}')
+        state = torch.load(model_filename)
```
Comment (KwiatkowskiML) [inline train.py:37]: state shadows name from outer scope

File: train.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        initial_epoch = state['epoch'] + 1
+        optimizer.load_state_dict(state['optimizer_state_dict'])
+        global_step = state['global_step']
+
+    loss_fn = nn.CrossEntropyLoss(ignore_index=tokenizer_src.token_to_id('[PAD]'), label_smoothing=0.1).to(device)
+
+    for epoch in range(initial_epoch, config['num_epochs']):
+        model.train()
+        batch_iterator = tqdm(train_dataloader, desc=f'Processing epoch {epoch:02d}')
```
Comment (KwiatkowskiML) [inline train.py:46]: tqdm is not imported

File: train.py [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        torch.save({
+            'epoch': epoch,
+            'model_state_dict': model.state_dict(),
+            'optimizer_state_dict': optimizer.state_dict(),
+            'global_step': global_step,
+        }, model_filename)
+
+if __name__ == '__main__':
+    warnings.filterwarnings('ignore')
+    config = get_config()
```
Comment (KwiatkowskiML) [inline train.py:89]: get_config is not imported
